pipeline_dp/dp_computations.py

```diff
@@ -39,9 +39,10 @@ def linf_sensitivity(self, metric):
             return self.max_contributions_per_partition * abs(
                 self.middle() - self.low)
         if metric == pipeline_dp.Metrics.VAR:
-            return self.max_contributions_per_partition * abs(
-                self.middle_squares() - self.low ** 2)
-        # TODO: add value for variance
+            max_contribution = max(self.low**2, self.high
+                                   **2) if self.low * self.high < 0 else abs(
+                                       self.high**2 - self.low**2)
+            return self.max_contributions_per_partition * max_contribution
```

Comment:
High-level concern: This method doesn't really make sense. There's no such thing as the linf sensitivity of a mean or a variance because each is created by post-processing several differently noised quantities. We're using these to mean "the sensitivity of a normalized sum" or "the sensitivity of a normalized sum of squares." I suspect the difference between the stated purpose of the method and what it's actually doing may cause confusion.

Also, calculating sensitivity like this makes things more error-prone because we have to keep this code in sync with the code that actually performs the aggregations. If we change the normalization process, we also have to modify the code here. Separating this calculation out makes it more likely that we'll forget to update both things.

I know this isn't a problem unique to this PR, but I think it becomes particularly bad for variance. Maybe now is an opportunity to improve things?

Reply:
Sure, thanks for suggesting this! I think it was clearer with count/sum when noise was added to only one value, but it definitely became confusing with mean/var. For now, I adapted the functions to directly set the Linf sensitivity for each mechanism. What do you think?

Reply:
This seems like a good solution. It's much clearer now. Thanks!